From f53000864e22033d82db7aa3bb91e70350ef203b Mon Sep 17 00:00:00 2001
From: Aryan Khare <aryakhar@qti.qualcomm.com>
Date: Wed, 1 Jul 2026 16:38:12 +0530
Subject: [PATCH] Fix OpenWrt Airties TLV regression caused by missing else
 branch

Restore the else() fallback in multi_vendor/CMakeLists.txt so that
OpenWrt (nl80211) builds compile the Airties TLV handler sources

Signed-off-by: Aryan Khare <aryakhar@qti.qualcomm.com>
---
 common/beerocks/multi_vendor/CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/common/beerocks/multi_vendor/CMakeLists.txt b/common/beerocks/multi_vendor/CMakeLists.txt
index 9c0ea63b3..725435c95 100644
--- a/common/beerocks/multi_vendor/CMakeLists.txt
+++ b/common/beerocks/multi_vendor/CMakeLists.txt
@@ -31,6 +31,9 @@ if (USE_PRPLMESH_WHM)
 elseif(TARGET_PLATFORM_TYPE STREQUAL "rdk")
     list(APPEND multi_vendor_sources ${MODULE_PATH}/onewifi/src/multi_vendor_airties_onewifi.cpp)
     list(APPEND multi_vendor_sources ${MODULE_PATH}/onewifi/src/tlvf_airties_utils_onewifi.cpp)
+else()
+    list(APPEND multi_vendor_sources ${MODULE_PATH}/nl80211/src/multi_vendor_airties_nl80211.cpp)
+    list(APPEND multi_vendor_sources ${MODULE_PATH}/nl80211/src/tlvf_airties_utils_nl80211.cpp)
 endif()
 
 # Build the library
-- 
2.34.1

